distributed/comm/ucx.py

```diff
@@ -60,8 +68,35 @@ def init_once():
                 "CUDA support with UCX requires Numba for context management"
             )
 
+        cuda_visible_device = int(
+            os.environ.get("CUDA_VISIBLE_DEVICES", "0").split(",")[0]
+        )
+        pre_existing_cuda_context = has_cuda_context()
```

Comment:
Would this lead to some sort of infinite recursion?

Reply:
I don't see how, where is the potential recursiveness here?

Reply:
Isn't `has_cuda_context` calling `init_once`?

Reply:
Oh, I now see why. You're actually mentioning the `init_once`. But it happens that both `nvml.py` and `ucx.py` have the `init_once` functions. `has_cuda_context` is calling that from the `nvml` module and not the one in `ucx` module. So recursion shouldn't happen.

Reply:
To be clear, `has_cuda_context` is calling [this `init_once`](https://github.com/dask/distributed/blob/983914f83b3eceee212e3c44fd80494c8e11828d/distributed/diagnostics/nvml.py#L15-L34)

Reply:
Oh got it. NVM 🤦‍♂️ Thanks for explaining that 😄